### ISSUE TITLE
Input linked to Kodi

### ISSUE THREAD

**Issue by zach-morris** [kodi-game/game.libretro.lutro#4]
Version tested: RetroPlayer 18 alpha1 2018-02-11

Launch game and then try to control it.  It appears as though the game controls are linked to Kodi command/controls.  For example when I press BUTTON [ 13 ] on my controller, the game is closed.  In this particular game I was able to control up/down/left/right but couldnt register any in game inputs (start or buttons did not seem to do anything)

```
21:35:04.026 T:123145304526848   DEBUG: CGUIInfoManager::SetCurrentGame(/Users/xxx/Library/Application Support/Kodi/userdata/addon_data/plugin.program.iarl/temp_iarl/Sienna.lutro)
...
21:35:04.090 T:123145310429184  NOTICE: [resources.lib.xbmcswift2b] Request for "/Emulator/Lutro_ZachMorris/1" matches rule for function "get_rom_page"
21:35:04.093 T:123145310429184   DEBUG: IARL: Loaded list cache from file for Lutro_ZachMorris
21:35:04.105 T:123145310429184   ERROR: NEWADDON Invalid Date Format ""
21:35:04.127 T:123145318555648   ERROR: Previous line repeats 1 times.
21:35:04.127 T:123145318555648   DEBUG: Skin Helper Service --> Kodi_Monitor: sender xbmc - method: Player.OnPlay  - data: {"item":{"type":"unknown"},"player":{"playerid":-1,"speed":1}}
21:35:04.137 T:123145310429184   ERROR: NEWADDON Invalid Date Format ""
21:35:04.139 T:123145318555648   DEBUG: Skin Helper Service --> Kodi_Monitor: sender xbmc - method: Unknown.OnChanged  - data: null
21:35:04.140 T:123145310429184   ERROR: NEWADDON Invalid Date Format ""
21:35:04.146 T:123145310429184   ERROR: Previous line repeats 5 times.
21:35:04.146 T:123145310429184    INFO: CPythonInvoker(34, /Users/xxx/Library/Application Support/Kodi/addons/plugin.program.iarl/addon.py): script successfully run
21:35:04.159 T:123145308819456   DEBUG: Thread BackgroundLoader start, auto delete: false
21:35:04.165 T:123145308819456   DEBUG: Thread BackgroundLoader 123145308819456 terminating
21:35:04.236 T:123145310429184    INFO: Python script stopped
21:35:04.236 T:123145310429184   DEBUG: Thread LanguageInvoker 123145310429184 terminating
21:35:04.406 T:140735305183232   DEBUG: ------ Window Deinit (DialogBusy.xml) ------
21:35:08.730 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: loading setting definitions
21:35:08.730 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: trying to load setting definitions from old format...
21:35:08.939 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:35:08.939 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:35:08.939 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:35:09.029 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:35:09.029 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:35:09.029 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:35:09.404 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:35:09.404 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:35:09.404 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:35:09.490 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:35:09.491 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:35:09.491 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:35:09.740 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:35:09.740 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:35:09.740 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:35:09.867 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:35:09.867 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:35:09.867 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:35:10.094 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:35:10.094 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:35:10.094 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:35:10.221 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:35:10.222 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:35:10.222 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:35:10.490 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:35:10.490 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:35:10.491 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:35:10.601 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:35:10.601 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:35:10.602 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:35:10.741 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:35:10.741 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:35:10.741 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:35:10.904 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:35:10.904 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:35:10.904 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:35:11.124 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:35:11.125 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:35:11.125 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:35:11.210 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:35:11.211 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:35:11.211 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:35:11.724 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:35:11.725 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:35:11.725 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:35:11.842 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:35:11.842 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:35:11.842 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:35:11.894 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:35:11.894 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:35:11.894 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:35:12.041 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:35:12.042 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:35:12.042 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:35:12.509 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" pressed
21:35:12.509 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (ignored)
21:35:12.509 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (handled)
21:35:12.621 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" released
21:35:12.622 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default released
21:35:12.724 T:123145315336192   DEBUG: Previous line repeats 1 times.
21:35:12.724 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:35:12.725 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:35:12.725 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:35:12.869 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:35:12.869 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:35:12.869 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:35:12.957 T:123145315872768   DEBUG: CRepositoryUpdater: busy playing. postponing scheduled update
21:35:13.010 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" pressed
21:35:13.010 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (ignored)
21:35:13.010 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (handled)
21:35:13.159 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" released
21:35:13.159 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default released
21:35:13.269 T:123145315336192   DEBUG: Previous line repeats 1 times.
21:35:13.269 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:35:13.270 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:35:13.270 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:35:13.374 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:35:13.374 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:35:13.375 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:35:13.485 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:35:13.485 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:35:13.485 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:35:13.590 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:35:13.590 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:35:13.590 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:35:13.694 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:35:13.694 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:35:13.694 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:35:13.735 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: loading setting definitions
21:35:13.736 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: trying to load setting definitions from old format...
21:35:13.806 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:35:13.806 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:35:13.806 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:35:13.891 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:35:13.892 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:35:13.892 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:35:13.942 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:35:13.943 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:35:13.943 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:35:14.189 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" pressed
21:35:14.189 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (ignored)
21:35:14.190 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (handled)
21:35:14.292 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" released
21:35:14.292 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default released
21:35:14.611 T:123145315336192   DEBUG: Previous line repeats 1 times.
21:35:14.611 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:35:14.611 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:35:14.611 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:35:14.740 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:35:14.740 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:35:14.741 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:35:14.932 T:123145315336192   DEBUG: BUTTON [ 13 ] on "PS(R) Gamepad" pressed
21:35:14.932 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.default pressed (ignored)
21:35:14.932 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.msx.joystick pressed (handled)
21:35:14.942 T:140735305183232   DEBUG: RetroPlayer: Closing file
```

**Comment by garbear** [comment]
Are you sure this is relevant to Lutro? `game.controller.msx.joystick` is only used for blueMSX and fMSX.

**Comment by zach-morris** [comment]
Yeah, I see that now.  I can say I was playing with MSX before trying a lutro game, but the log was created when a Lutro game was loaded.

**Comment by garbear** [comment]
In the code I emulate controllers connecting by connecting retroplayer to peripherals and input. I think lifetime issues between the two were causing a race condition when reconnecting the controller. The fix was 1 line and made it into the 2018-02-15 build.

**Comment by zach-morris** [comment]
Input looks like its fixed in 18.0-ALPHA1 Git:20180215-276e1a4
